packages/sync-actions/src/utils/create-build-actions.js

```diff
@@ -1,13 +1,23 @@
-export default function createBuildActions (diff, doMapActions) {
+function applyOnBeforeDiff (before, now, fn) {
+  return fn && typeof fn === 'function' ? fn(before, now) : [before, now]
+}
+
+export default function createBuildActions (
+  differ, doMapActions, onBeforeDiff,
+) {
   return function buildActions (now, before, options = {}) {
     if (!now || !before)
       throw new Error('Missing either `newObj` or `oldObj` ' +
         'in order to build update actions')
 
-    // diff 'em
-    const diffed = diff(before, now)
+    const [
```

Comment:
This function feels generally flaky to me. It shuffles the arguments of `new` and `before` around quite a bit and I wonder if that's the intention or actually a bug which we then work around when consuming the diff.